File: aspnet-in-practice/05/CS/Display/App_Code/ObjectContextModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using ObjectModel;

public class ObjectContextModule: IHttpModule
{
	public void Dispose()
	{
		// nothing
	}

	public void Init(HttpApplication context)
	{
		context.BeginRequest += new EventHandler(context_BeginRequest);
		context.EndRequest += new EventHandler(context_EndRequest);
		context.Error += new EventHandler(context_EndRequest);
	}

	void context_EndRequest(object sender, EventArgs e)
	{
		if (HttpContext.Current.Items["ObjectContext"] != null)
			(HttpContext.Current.Items["ObjectContext"] as NorthwindEntities).Dispose();
	}

	void context_BeginRequest(object sender, EventArgs e)
	{
		HttpContext.Current.Items["ObjectContext"] = new NorthwindEntities();
	}
}

File: aspnet-in-practice/05/VB/Display/App_Code/ObjectContextModule.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports ObjectModel

Public Class ObjectContextModule
	Implements IHttpModule

	Public Sub Dispose() Implements IHttpModule.Dispose
		' nothing
	End Sub

	Public Sub Init(ByVal context As HttpApplication) Implements IHttpModule.Init
		AddHandler context.BeginRequest, AddressOf context_BeginRequest
		AddHandler context.EndRequest, AddressOf context_EndRequest
		AddHandler context.Error, AddressOf context_EndRequest
	End Sub

	Private Sub context_EndRequest(ByVal sender As Object, ByVal e As EventArgs)
		If HttpContext.Current.Items("ObjectContext") IsNot Nothing Then
			TryCast(HttpContext.Current.Items("ObjectContext"), NorthwindEntities).Dispose()
		End If
	End Sub

	Private Sub context_BeginRequest(ByVal sender As Object, ByVal e As EventArgs)
		HttpContext.Current.Items("ObjectContext") = New NorthwindEntities()
	End Sub
End Class